src/drivers/mct/main/cime_comp_mod.F90

```diff
@@ -4467,4 +4542,42 @@ subroutine cime_run_write_restart(drv_pause, write_restart, drv_resume)
 
   end subroutine cime_run_write_restart
 
+!----------------------------------------------------------------------------------
+
+  subroutine cime_write_performance_checkpoint(output_ckpt, ckpt_filename, &
+                                               ckpt_mpicom)
+
+    !----------------------------------------------------------
+    ! Checkpoint performance data
+    !----------------------------------------------------------
+
+    logical, intent(in)             :: output_ckpt
+    character(len=*), intent(in)    :: ckpt_filename
+    integer, intent(in)             :: ckpt_mpicom
+
+103 format( 5A )
+104 format( A, i10.8, i8)
+
+    call t_adj_detailf(+1)
+
+    call t_startf("sync1_tprf")
+    call mpi_barrier(ckpt_mpicom,ierr)
```

Comment:
This code introduces an mpi_barrier to the cime_run loop - it could have serious performance implications.  In other cases timing barriers are inside ``` if (barrier_alarm)  ```  code blocks.  It seems like this should be the same.

Reply:
@matthewhoffman was that barrier needed?

Reply:
@rljacob , I'm not familiar with this code - it is not part of the PR from me.

Reply:
@worleyph made this change I think (7b1e7442 in E3SM)

Reply:
This same logic existed in-line previously. I just encapsulated it in a subroutine to cut down on duplicate code. It is called in one new location, just before the start of the run loop (but outside), immediately preceding the existing RUN_LOOP_BSTART barrier (which is annoying to have both, but seemed preferable to the other options I considered). 
It is executed immediately after initialization (so once - a new feature, and the point the PR), after the end of the first day (also once, has been this way for a long time), and whenever tprof_alarm is true (also for a long time). The barriers also occur during cime_final.

Reply:
The communicator on the barrier changed. It was call mpi_barrier(mpicom_GLOID,ierr)

Reply:
@worleyph thanks for the explanation, that makes sense - but since mpicom_GLOID is global anyway can we remove it as a function argument and just use the global variable?  I think that
renaming the variable creates confusion.

Reply:
There has always been the possibility that components would write out there own checkpoints (or even do their timing independently). We have never supported that, but the design has always been in the back of my head, thus the use of a formal parameter and not hardwiring mpicom_GLOID. This is a subroutine, and the communicator is clearly a passed parameter. I don't find this confusing :-). You just have to check the call sites? So... my opinion is to leave it as is. However, do what you feel that you need to do.

Reply:
Actually, I'd strongly prefer that this remain a formal parameter. There are a lot of comunicators floating around in cime_comp_mod.F90. It seems a strange design to hardwire GLOID in a subroutine that is otherwise pretty generic.

Reply:
I see why we might want to change it back to GLOID.  If we want to search the code for everywhere a barrier is called on GLOID, it would be missed.  Its not obvious from the name of the function that a barrier is going to called inside it.

Reply:
t_drvstartf, t_drvstopf also has the (optional) parameter to specify communicator to be used in a barrier, though it is t_barrierf and so not normally enabled. It also never uses GLOID (currently), but rather CPLID and omp(eci)%mpicom_compid, and mpicom_barrier (as called from component_exch, so even more once removed).

seq_domain_check_grid also has a barrier (and an allreduce) using an mpicom formal parameter. 

Could make this an optional parameter that defaults to GLOID if the parameter is not specified, so grep would pick it up? But, aince all of the calls (all two of them) are local to the file cime_comp_mod.F90 at the moment, would not be hard to hunt down.

In general, if there is a need to identify all barrier (and allreduce?) calls using GLOID, I wouldn't depend on grep without further investigation of all barrier and allreduce calls, unless we eliminate all communicator parameters. So, my preference is the same, to leave as it, but will make changes if people request it (or you can make the changes yourself).

Reply:
Telecon: we are fine with this as is.